src/encoded/types/shared_calculated_properties.py

```diff
@@ -482,30 +482,46 @@ class CalculatedReplicationType:
         "description": "Calculated field that indicates the replication model",
         "type": "string"
     })
-    def replication_type(self, request, replicates=None, assay_term_name=None):
+    def replication_type(self, request, original_files=None, replicates=None, assay_term_name=None):
         # ENCD-4251 loop through replicates and select one replicate, which has
         # the smallest technical_replicate_number, per biological replicate.
         # That replicate should have a libraries property which, as calculated
         # in replicate.libraries (ENCD-4251), should have collected all
         # possible technical replicates belong to the biological replicate.
         # TODO: change this once we remove technical_replicate_number.
+        # replication_type for experiment without data files is not defined
+        if not original_files:
+            return None
+        # replication_type for experiment without replicates is not defined
+        if not replicates:
+            return None
+        
+        excluded_statuses = ['uploading', 'content error', 'upload failed', 'deleted', 'replaced']
+        replicates_with_files = set()
+        for file_id in original_files:
+            file_object = request.embed(file_id, '@@object')
+            if file_object['status'] not in excluded_statuses:
```

Comment:
There's this paths_filtered_by_status function that you might want to use: https://github.com/ENCODE-DCC/encoded/blob/88169b961a82a1d45772772dabd07eff8e82cf5d/src/encoded/types/base.py#L117

Original_files and files use it e.g.: https://github.com/ENCODE-DCC/encoded/blob/88169b961a82a1d45772772dabd07eff8e82cf5d/src/encoded/types/dataset.py#L127

Reply:
updated, but not sure that is what you had in mind @keenangraham